e2e-tests/commands/assert-summary-list-row.js

```diff
@@ -0,0 +1,48 @@
+import { LINKS, DEFAULT } from '../content-strings';
+
+/**
+ * assertSummaryListRow
+ * Assert a summary list row.
+ * @param {Object} summaryList list cypress selectors from pages/partials.
+ * @param {String} Field ID in the summary list to assert
+ * @param {String} expectedKey expected key of the row
+ * @param {String} expectedValue expected value of the row
+ * @param {String} expectedChangeLinkText expected change link text - either add/change or should not exist
+ */
+const assertSummaryListRow = (summaryList, fieldId, expectedKey, expectedValue, expectedChangeLinkText) => {
+  const row = summaryList.field(fieldId);
+
+  cy.assertSummaryListRowKey(summaryList, fieldId, expectedKey);
+
+  /**
+   * if value exists then checkText
+   * else check if dash
+   */
+  if (expectedValue) {
+    cy.assertSummaryListRowValue(summaryList, fieldId, expectedValue);
+  } else {
+    cy.assertSummaryListRowValue(summaryList, fieldId, DEFAULT.EMPTY);
+  }
+
+  /**
+   * if change link text, if there is an expected value, then should contain change
+   * else if no expected value, should say add
+   * if no expectedChangeLinkText, then should not have changeLink
+   */
+  if (expectedChangeLinkText) {
+    if (expectedValue) {
+      cy.assertSummaryListRowChangeLink(summaryList, fieldId, `${LINKS.CHANGE} ${expectedChangeLinkText}`);
+
+      cy.checkText(
+        row.changeLink(),
+        `${LINKS.CHANGE} ${expectedChangeLinkText}`,
+      );
+    } else {
+      cy.assertSummaryListRowChangeLink(summaryList, fieldId, `${LINKS.ADD} ${expectedChangeLinkText}`);
+    }
+  } else {
+    cy.assertSummaryListRowChangeLink(summaryList, fieldId);
+  }
+};
```

Comment:
Looks like we don't need `checkText` on line 36 as `assertSummaryListRowChangeLink` does this?

This feels strange to read as it has a `expectedChangeLinkText, param, but then different text is checked ("add"/"change"?) Could it just be this?

```js
if (expectedChangeLinkText) {
  cy.assertSummaryListRowChangeLink(summaryList, fieldId, expectedChangeLinkText);
} else {
  cy.assertSummaryListRowChangeLink(summaryList, fieldId);
}
```

Also looks 4 params are being passed to `assertSummaryListRowChangeLink` but it only uses 3 of them.